app/support/auto_expire.rb

```diff
@@ -0,0 +1,44 @@
+class AutoExpire
+  def perform
+    order_details.each do |od|
+      od.transaction do
+        expire_reservation(od)
+      end
+    end
+  end
+
+  private
+
+  def order_details
+    purchased_active_order_details | non_reservation_order_details
```

Comment:
This feels a little weird to me. I feel like we might be conflating two ideas here. We could do both of them in the same rake task, and they would likely share a bit of code, but I think it would make sense to have one that ends reservation-only OrderDetails and one that ends the un-logged-out instruments.